pkg/controller/autoscaler/horizontalpodautoscaler_controller.go

```diff
@@ -37,6 +37,10 @@ import (
 const (
 	heapsterNamespace = "kube-system"
 	heapsterService   = "monitoring-heapster"
+
+	// Usage shoud go below downscaleTolerance of the target before we start downscale the pods.
+	// TODO: make it a flag or HPA spec element.
+	downscaleTolerance = 0.9
```

Comment:
After some consideration, I think this approach is not symmetric: we scale down at 0.9 and scale up at 1.0. Maybe we should make it more similar: 0.95 for scale-down and 1.05 for scale up?